src/Disks/StoragePolicy.cpp

```diff
@@ -105,9 +99,16 @@ StoragePolicy::StoragePolicy(String name_, Volumes volumes_, double move_factor_
 
     for (size_t i = 0; i < volumes.size(); ++i)
     {
-        if (volumes_names.find(volumes[i]->getName()) != volumes_names.end())
-            throw Exception("Volumes names must be unique in storage policy " + backQuote(name) + " (" + volumes[i]->getName() + " duplicated).", ErrorCodes::UNKNOWN_POLICY);
-        volumes_names[volumes[i]->getName()] = i;
+        if (volume_index_by_volume_name.find(volumes[i]->getName()) != volume_index_by_volume_name.end())
+            throw Exception("Volume names must be unique in storage policy " + backQuote(name) + " (" + volumes[i]->getName() + " is duplicated).", ErrorCodes::UNKNOWN_POLICY);
+        volume_index_by_volume_name[volumes[i]->getName()] = i;
+        for (const auto & disk : volumes[i]->getDisks())
+        {
+            const String & disk_name = disk->getName();
+            if (volume_index_by_disk_name.find(disk_name) != volume_index_by_disk_name.end())
+                throw Exception("Disk names must be unique in storage policy " + backQuote(name) + " (" + disk_name + " is duplicated)", ErrorCodes::EXCESSIVE_ELEMENT_IN_CONFIG);
+            volume_index_by_disk_name[disk_name] = i;
+        }
```

Comment:
Looks like copy-paste, I guess can be easily fixed

Reply:
+